src/snarks/groth16/mpc/chacha_rng.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef __ZETH__SNARKS_GROTH16_MPC_CHACHA_RNG_HPP__
+#define __ZETH__SNARKS_GROTH16_MPC_CHACHA_RNG_HPP__
+
+#include <cstddef>
+#include <cstdint>
+
+namespace libzeth
+{
+
+// Random number generator matching the implementation used by zcash
+// powersoftau and phase2.
+// Usage:
+//   https://github.com/clearmatics/powersoftau
+//   (See hash_to_g2 function)
+// Implementation:
+//   https://github.com/rust-random/rand/blob/master/rand_chacha/src/chacha.rs
+//   (See description comment, in particular word layout)
```

Comment:
Let's use `///` type of comments as this is a useful comment for the documentation

Reply:
Switched to `///`